controller/users.go

```diff
@@ -215,9 +215,43 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 func (c *UsersController) List(ctx *app.ListUsersContext) error {
 	return application.Transactional(c.db, func(appl application.Application) error {
 		var err error
-		var users []*account.User
 		var result *app.UserArray
-		users, err = appl.Users().List(ctx.Context)
+
+		if ctx.Username != nil {
+			// From a data model perspective, we are querying by identity ( and not user )
+			identities, err := appl.Identities().Query(account.IdentityFilterByUsername(*ctx.Username))
```

Comment:
@hectorj2f 
> But if not defined, we list all the users. That is weird part in my opinion.

If no filters/params are passed, we return all users , so as to not break the existing API behaviour :( 

> Wouldn't this be a Show operation ?  I understand username and email will return one user. 
If there are 2 identities for the same user, the search result would have 2 data items.
I've converted params into filter params now. I feel it fits fine into `List` unless you suggest we put it somewhere else :)


@alexeykazakov :
Defaulted to kc type : https://github.com/almighty/almighty-core/pull/1196/commits/423433d79d7d3a3e0d79a19197f31dded098c745#diff-da062b312519ee2a61dd669f1ec964d4R244

Didn't expose a filter for `providerType` in the REST API for now, but the controller code has been designed it a way that we can add it anytime we want without any extra effort.